jcl/src/java.base/share/classes/openj9/internal/tools/attach/target/TargetDirectory.java

```diff
@@ -130,33 +135,40 @@ static String createMyDirectory(String myVmId, boolean preserveId) throws IOExce
 	 */
 	static boolean deleteMyFiles() {
 		if (LOGGING_DISABLED != loggingStatus) {
-			IPC.logMessage("deleting my files"); //$NON-NLS-1$
+			IPC.logMessage("deleting my files: attachInfo files and attachNotificationSync files."); //$NON-NLS-1$
 		}
 
 		if ((null != advertisementFileObject) && advertisementFileObject.delete()) {
+			IPC.logMessage("deleted ", advertisementFileObject.getAbsolutePath()); //$NON-NLS-1$
 			advertisementFileObject = null;		
 		} else {
 			return false;
 		}
 		if ((null != syncFileObject) && syncFileObject.delete()) {
+			IPC.logMessage("deleted ", syncFileObject.getAbsolutePath()); //$NON-NLS-1$
 			syncFileObject = null;
 		} else {
 			return false;
 		}
+		if (LOGGING_DISABLED != loggingStatus) {
+			IPC.logMessage("deleted my files with success."); //$NON-NLS-1$
+		}
 		return true;
 	}
 
 	/**
 	 * Remove this VM's target directory and the files within it
 	 * @param directoryEmpty set to true if the directory is supposedly empty
+	 * @vmId target VM ID
 	 */
-	static void deleteMyDirectory(boolean directoryEmpty) {
+	static void deleteMyDirectory(boolean directoryEmpty, String vmId) {
 		if (LOGGING_DISABLED != loggingStatus) {
-			IPC.logMessage("deleting my directory "); //$NON-NLS-1$
+			IPC.logMessage("deleting my directory with given vmId: ", vmId); //$NON-NLS-1$
 		}		
 		/* try the fast path for deleting files and directories. try the heavyweight path if something fails */
 		if ((!directoryEmpty && !deleteMyFiles()) || (null == targetDirectoryFileObject) || !targetDirectoryFileObject.delete()) {
-			deleteTargetDirectory(AttachHandler.getVmId());	
```

Comment:
We will have to check if `vmId` is empty, if so, set it to `AttachHandler.getVmId()` to keep current behaviour.

Reply:
As I discussed before,  looks no need to keep both
`deleteMyDirectory(boolean directoryEmpty)`
and `static void deleteMyDirectory(boolean directoryEmpty, String vmId)`

So, current p[roposed change looks fine, what's issue here?

Reply:
Please see my suggestion below.